src/KOKKOS/pair_eam_kokkos.cpp

```diff
@@ -799,6 +843,232 @@ void PairEAMKokkos<DeviceType>::operator()(TagPairEAMKernelC<NEIGHFLAG,NEWTON_PA
   EV_FLOAT ev;
   this->template operator()<NEIGHFLAG,NEWTON_PAIR,EVFLAG>(TagPairEAMKernelC<NEIGHFLAG,NEWTON_PAIR,EVFLAG>(), ii, ev);
 }
+/* ---------------------------------------------------------------------- */
+
+////Specialisation for Neighborlist types Half, HalfThread, Full
+template<class DeviceType>
+template<int EFLAG>
+KOKKOS_INLINE_FUNCTION
+void PairEAMKokkos<DeviceType>::operator()(TagPairEAMKernelAB<EFLAG>,
+		                     const typename Kokkos::TeamPolicy<DeviceType>::member_type& team_member,
+				     EV_FLOAT& ev) const {
+  int ii = team_member.league_rank()*team_member.team_size() + team_member.team_rank();
```

Comment:
There's never a check of if `ii < imax`, so if your number of atoms doesn't divide evenly into the `team_size()` (hard-coded as 1024 in the initial commit) you'll get some nasty out-of-bounds accesses. I see that you use all of the threads in a team as part of the load into scratch memory, so you could put the `ii < imax` check after that.